pkg/config/config.go

```diff
@@ -106,6 +106,17 @@ type ClusterNetworkEntry struct {
 	CIDR string `json:"cidr,omitempty"`
 }
 
+type DNS struct {
+	// baseDomain is the base domain of the cluster. All managed DNS records will
+	// be sub-domains of this base.
+	//
+	// For example, given the base domain `openshift.example.com`, an API server
+	// DNS record may be created for `cluster-api.openshift.example.com`.
+	//
+	// Once set, this field cannot be changed.
```

Comment:
Do we need another ticket to protect against the value changing?

Reply:
I believe that this is a comment inherited from the OCP config API: https://github.com/openshift/api/blob/master/config/v1/types_dns.go#L27-L34

We might want to update it if it doesn't apply to MicroShift

Reply:
It does apply to MicroShift, too, I think @dhellmann 's concern was more where/how this is enforced.

Reply:
We may want to enforce against changes, I agree.